src/hooks/useUser.ts

```diff
@@ -1,29 +1,44 @@
-import { User, UseUserReturnType } from '@/interfaces';
-import { useSession } from 'next-auth/react';
 import { useState, useEffect } from 'react';
+import { useSession } from 'next-auth/react';
+import { User, UseUserReturnType } from '@/interfaces';
 
 const useUser = (): UseUserReturnType => {
   const { data: session, status } = useSession();
-  const [user, setUser] = useState<User>(session?.user as User);
+  const [user, setUser] = useState<User | null>(null);
   const [isAuth, setIsAuth] = useState(false);
+  const [isOnboarded, setIsOnboarded] = useState<boolean | undefined>(
```

Comment:
Making it complicated for no reason. Either user is onboarded or not. Why undefined.